src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/SubjectAccessRequestRepository.kt

```diff
@@ -0,0 +1,96 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity
+
+import org.postgresql.util.PGobject
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import org.springframework.stereotype.Repository
+import java.time.LocalDateTime
+
+@Repository
+class SubjectAccessRequestRepository(
+  val jdbcTemplate: NamedParameterJdbcTemplate,
+) {
+
+  fun getApprovedPremisesApplicationsJson(
+    crn: String?,
+    nomsNumber: String?,
+    startDate: LocalDateTime?,
+    endDate: LocalDateTime?,
+  ): String {
+    val result = jdbcTemplate.queryForMap(
+      """
+        select 
+        	json_agg(app) as json
+        from (
+        select
+          a.id,
+          apa.name,
+        	a.crn,
+          a.noms_number,
+        	a."data",
+        	a."document",
+        	a.created_at,
+        	a.submitted_at,
+          created_by_user.name as created_by_user,
+        	app_user."name" as application_user_name ,
+        	apa.event_number,
+        	apa.is_womens_application,
+        	apa.offence_id,
+        	apa.conviction_id,
+        	apa.risk_ratings,
+        	apa.release_type,
+        	apa.arrival_date,
+        	apa.is_withdrawn,
+        	apa.withdrawal_reason,
+        	apa.other_withdrawal_reason ,
+        	apa.is_emergency_application,
+        	apa.target_location ,
+        	apa.status,
+        	apa.inmate_in_out_status_on_submission,
+        	apa.sentence_type,
+        	apa.notice_type,
+        	apa.ap_type,
+        	case_manager."name" as case_manager_name,
+        	apa.case_manager_is_not_applicant
```

Comment:
could you please make the indentation above consistent as it helps see the 'shape' of the SQL

Reply:
Yep that's a bit odd. will fix.